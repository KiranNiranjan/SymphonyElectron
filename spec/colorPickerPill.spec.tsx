import { shallow } from 'enzyme';
import * as React from 'react';
import ColorPickerPill from '../src/renderer/components/color-picker-pill';

const defaultProps = {
  availableColors: [
    { rgbaColor: 'rgba(0, 142, 255, 0.64)' },
    { rgbaColor: 'rgba(38, 196, 58, 0.64)' },
    { rgbaColor: 'rgba(246, 178, 2, 0.64)' },
    { rgbaColor: 'rgba(233, 0, 0, 0.64)' },
  ],
  onChange: jest.fn(),
};

describe('<ColorPickerPill/>', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<ColorPickerPill {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onChange when clicked on a color dot', () => {
    const wrapper = shallow(<ColorPickerPill {...defaultProps} />);
    wrapper
      .find('[data-testid="colorDot rgba(233, 0, 0, 0.64)"]')
      .simulate('click');
    expect(defaultProps.onChange).toHaveBeenCalledTimes(1);
  });

  it('should render chosen dots as larger', () => {
    const chosenColorProps = {
      availableColors: [
        { rgbaColor: 'rgba(0, 0, 0, 0.64)', chosen: true },
        { rgbaColor: 'rgba(233, 0, 0, 0.64)' },
      ],
      onChange: jest.fn(),
    };
    const wrapper = shallow(<ColorPickerPill {...chosenColorProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render outlined dot if provided in props', () => {
    const outlinedColorProps = {
      availableColors: [
        { rgbaColor: 'rgba(246, 178, 2, 1)' },
        { rgbaColor: 'rgba(255, 255, 255, 1)', outline: 'rgba(0, 0, 0, 1)' },
      ],
      onChange: jest.fn(),
    };
    const wrapper = shallow(<ColorPickerPill {...outlinedColorProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
