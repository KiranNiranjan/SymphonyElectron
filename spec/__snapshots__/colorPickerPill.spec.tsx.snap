// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorPickerPill/> should render chosen dots as larger 1`] = `
<div
  className="color-picker"
>
  <button
    className="enclosing-circle"
    data-testid="colorDot rgba(0, 0, 0, 0.64)"
    key="rgba(0, 0, 0, 0.64)"
    onClick={[Function]}
    tabIndex={0}
  >
    <div
      className="color-dot"
      style={
        Object {
          "background": "rgba(0, 0, 0, 0.64)",
          "border": undefined,
          "cursor": "pointer",
          "height": "24px",
          "width": "24px",
        }
      }
    />
  </button>
  <button
    className="enclosing-circle"
    data-testid="colorDot rgba(233, 0, 0, 0.64)"
    key="rgba(233, 0, 0, 0.64)"
    onClick={[Function]}
    tabIndex={0}
  >
    <div
      className="color-dot"
      style={
        Object {
          "background": "rgba(233, 0, 0, 0.64)",
          "border": undefined,
          "cursor": "pointer",
          "height": "8px",
          "width": "8px",
        }
      }
    />
  </button>
</div>
`;

exports[`<ColorPickerPill/> should render correctly 1`] = `
<div
  className="color-picker"
>
  <button
    className="enclosing-circle"
    data-testid="colorDot rgba(0, 142, 255, 0.64)"
    key="rgba(0, 142, 255, 0.64)"
    onClick={[Function]}
    tabIndex={0}
  >
    <div
      className="color-dot"
      style={
        Object {
          "background": "rgba(0, 142, 255, 0.64)",
          "border": undefined,
          "cursor": "pointer",
          "height": "8px",
          "width": "8px",
        }
      }
    />
  </button>
  <button
    className="enclosing-circle"
    data-testid="colorDot rgba(38, 196, 58, 0.64)"
    key="rgba(38, 196, 58, 0.64)"
    onClick={[Function]}
    tabIndex={0}
  >
    <div
      className="color-dot"
      style={
        Object {
          "background": "rgba(38, 196, 58, 0.64)",
          "border": undefined,
          "cursor": "pointer",
          "height": "8px",
          "width": "8px",
        }
      }
    />
  </button>
  <button
    className="enclosing-circle"
    data-testid="colorDot rgba(246, 178, 2, 0.64)"
    key="rgba(246, 178, 2, 0.64)"
    onClick={[Function]}
    tabIndex={0}
  >
    <div
      className="color-dot"
      style={
        Object {
          "background": "rgba(246, 178, 2, 0.64)",
          "border": undefined,
          "cursor": "pointer",
          "height": "8px",
          "width": "8px",
        }
      }
    />
  </button>
  <button
    className="enclosing-circle"
    data-testid="colorDot rgba(233, 0, 0, 0.64)"
    key="rgba(233, 0, 0, 0.64)"
    onClick={[Function]}
    tabIndex={0}
  >
    <div
      className="color-dot"
      style={
        Object {
          "background": "rgba(233, 0, 0, 0.64)",
          "border": undefined,
          "cursor": "pointer",
          "height": "8px",
          "width": "8px",
        }
      }
    />
  </button>
</div>
`;

exports[`<ColorPickerPill/> should render outlined dot if provided in props 1`] = `
<div
  className="color-picker"
>
  <button
    className="enclosing-circle"
    data-testid="colorDot rgba(246, 178, 2, 1)"
    key="rgba(246, 178, 2, 1)"
    onClick={[Function]}
    tabIndex={0}
  >
    <div
      className="color-dot"
      style={
        Object {
          "background": "rgba(246, 178, 2, 1)",
          "border": undefined,
          "cursor": "pointer",
          "height": "8px",
          "width": "8px",
        }
      }
    />
  </button>
  <button
    className="enclosing-circle"
    data-testid="colorDot rgba(255, 255, 255, 1)"
    key="rgba(255, 255, 255, 1)"
    onClick={[Function]}
    tabIndex={0}
  >
    <div
      className="color-dot"
      style={
        Object {
          "background": "rgba(255, 255, 255, 1)",
          "border": "solid 1px rgba(0, 0, 0, 1)",
          "cursor": "pointer",
          "height": "6px",
          "width": "6px",
        }
      }
    />
  </button>
</div>
`;
